client/lib/models/exercises/exercise.dart

```diff
@@ -7,12 +7,12 @@ part 'exercise.g.dart';
 @JsonSerializable()
 class Exercise {
   const Exercise({
-    required final this.id,

```

Comment:
Why did you remove the `final` specifier everywhere? This obviously also applies to other files.